dev/release_notes.readme.md

```diff
@@ -0,0 +1,102 @@
+
+Note that a file with release notes that is expected to be **read** must be **short**; other tools would be needed for **evaluating** the release notes (searching/filtering facilities according to various aspects).
+
+Besides the list of relevant pull requests, the release notes contain also the lists of changed GAP packages:
+- omitted packages (which had been distributed with the previous GAP version but are not distributed with the current one),
+- changed packages (which are distributed with both the previous and the current GAP version, but with different package versions),
+- new packages (which had not been distributed with the previous GAP version but are distributed with the current one).
+
+For each package in question, there is one entry that shows package name (with a link to the package homepage), version, authors and abstract.
+
+Note that the information about packages cannot be extracted from pull requests. It depends on the archives of packages that are distributed with GAP.
+(It would be easy to put this information together if there woud be a text file for each archive that lists which package versions are contained in the archive.)
+
+
+## How are release notes created from the github pull requests?
+
+The main idea is to use the *labels* that are assigned to the pull requests.
+Currently the following labels are regarded as relevant for the creation of release notes.
+
+- Pull requests with the label `release notes: not needed` shall **not** be listed in the release notes.
+  Thus the release notes are composed from all those merged pull requests that belong to the release in question (are assigned to the release branch or have a `backport-to-...-DONE` label for the release) AND do NOT have the label `release notes: not needed`.
+
+- For pull requests with the label `release notes: use title`, the text for the release notes is given by the title of the pull request.
+  For all other pull requests, the text for the release notes can be extracted from the pull request body.
+  (The pull request template contains `## Text for release notes`, thus the relevant text is expected there; but where is the end of this text? We have to define a convention to mark the release notes text, then this text can be extracted automatically.)
+
+- Up to now, also the labels `release notes: needed` and `release notes: added` have been used.
+  The first one means that the text for release notes can be found in the body and has not yet been added to `CHANGES.md`, the second one means that the text has been added to `CHANGES.md`.
+  I think that these two labels are obsolete in the new workflow where one can automatically create the current state of the `CHANGES.md` section about the forthcoming release **at any time**.
```

Comment:
This is a good idea. If nobody objects then we should proceed like this.

Reply:
I don't understand how the new process renders these labels obsolete. How do you distinguish between a PR that nobody has looked at yet from one where the title alone is not suitable for the release notes or from one that is not supposed to go into the release notes?

Also, how would the script distinguish between PRs that are already in the release notes of a previous release from those which are not? One could try to use a date for this, but in the past, this didn't work well (granted, that was before we used the policy of always merging into master and then doing backports and using the backport labels, so perhaps *nowadays* it works -- but I'd be at least careful about making this assumption...

Reply:
@fingolfin Concerning the distinction of pull requests w.r.t. release notes via labels, I think that what you ask for corresponds to `release notes: approved`, meaning that "someone has looked at this pull request, and it is ready for being processed automatically". (One could even evaluate whether the approval was added by the same person who added the label that marks the pull request as relevant for the release notes.)

Concerning the assignment of pull requests to releases, it will be possible to check that a pull request is not mentioned in the (Json format?) text files that list the release notes of earlier releases.